src/parser.h

```diff
@@ -43,6 +43,7 @@ struct cmark_parser {
   /* Options set by the user, see the Options section in cmark.h */
   int options;
   bool last_buffer_ended_with_cr;
+  cmark_llist *syntax_extensions;
```

Comment:
I'd rather have a generic container than intrusive linked lists where one has to reimplement standard iterating and freeing functions.